pathgen/pathgen.go

```diff
@@ -1007,6 +1009,13 @@ func generateDirectorySnippet(directory *ygen.ParsedDirectory, directories map[s
 			})
 		}
 	}
+
+	if directory.Type == ygen.List && len(directory.ListKeys) == 0 {
+		return snippets, errs
+	}
```

Comment:
nit: keep `ChildConstructors: methodBuf.String(),`, change this if statement to
```
// Since it is not possible for gNMI to refer to individual nodes underneath an unkeyed list, prevent constructing paths below a keyless list.
snippets[nonLeafIdx].ChildConstructors = ""
```